api/api.go

```diff
@@ -373,11 +382,40 @@ func (a *API) Recover() (map[int]*[32]byte, error) {
 // Returns:
 //   - *data.RestorationStatus: Status of the restoration process if successful
 //   - error: nil on success, unauthorized error if not authorized, or
-//     wrapped error on request/parsing failure
+//     wrapped error on request-parsing failure
 //
 // Example:
 //
 //	status, err := api.Restore(shardPtr)
 func (a *API) Restore(index int, shard *[32]byte) (*data.RestorationStatus, error) {
 	return operator.Restore(a.source, index, shard)
 }
+
+// CipherEncrypt encrypts using selected mode.
+// For stream mode, set mode=cipher.ModeStream and provide reader and contentType.
+// For JSON mode, set mode=cipher.ModeJSON and provide plaintext and algorithm.
+// Returns either SecretEncryptResponse (JSON mode) or ciphertext bytes (stream mode).
+func (a *API) CipherEncrypt(
+	mode cipher.Mode,
+	reader io.Reader,
+	contentType string,
+	plaintext []byte,
+	algorithm string,
+) (*reqres.SecretEncryptResponse, []byte, error) {
+	return cipherEncryptFunc(a.source, mode, reader, contentType, plaintext, algorithm)
+}
```

Comment:
@parlakisik I know that we initially decided to have a single method and switch the behavior based on the contentType parameter, but now I realize this results in passing a bunch of `nil` placeholders just to satisfy the "stream mode" contract.

I think we'll be better off if we have separate methods for stream and JSON/rest.